opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -688,9 +707,15 @@ def set_attributes(
                 return
 
             for key, value in attributes.items():
-                if not _is_valid_attribute_value(value):
+                valid, cleaned = _clean_attribute_value(
+                    value, self._limits.max_attribute_size
+                )
+                if not valid:
                     continue
 
+                if cleaned is not None:
+                    value = cleaned
+
                 if not key:
```

Comment:
Is the below validations still needed if `_clean_attribute_value` already does so?

Reply:
Yes, that was the intention. Fixed now.